lib/indico.js

```diff
@@ -12,36 +12,93 @@ var public_api = 'http://apiv2.indico.io/'
 indico.apiKey = false;
 indico.cloud = false;
 
-function service(name, privateCloud, apiKey) {
+function service(name, privateCloud, apiKey, apis) {
+  /* given a set of parameters, returns the proper url for the REST api*/
   if (privateCloud) {
     var prefix = 'http://' + privateCloud + '.indico.domains/';
-    var url = prefix + name.toLowerCase();
+    var url = prefix + name;
   } else {
-    var url = public_api + name.toLowerCase();
+    var url = public_api + name;
   }
   url = url + "?key=" + apiKey;
+  if (apis) {
+    url = url + "&apis=" + apis.join();
+  }
   return url
 }
 
+function formatResults(results, nameMap) {
+  var formatted_results = {};
+  for (result in results) {
+    if ('results' in results[result]) {
+      formatted_results[nameMap[result]] = results[result]['results'];
+    } else {
+      formatted_results[nameMap[result]] = results[result];
+    }
+  }
+  return formatted_results;
+}
+
+function addKeywordArguments(body, config) {
+  // pass along additional keyword args in JSON body
+  for (var key in config) {
+    if (key !== 'apiKey' && key !== 'privateCloud') {
+      body[key] = config[key];
+    } 
+  }
+  return body;
+}
+
+function standardizeApiNames(config, nameMap) {
+  // standardization for multi-api support
+  for (var key in config) {
+    if (key === 'apis') {
+      var apis = config['apis'];
+      for (apiName in apis) {
+        standardized = apis[apiName].toLowerCase();
+        nameMap[standardized] = apis[apiName]
+        apis[apiName] = standardized;
+      }
+      return apis
+    }
+  }
+  return false;
+}
 
-var make_request = function (name, data, config) {
+function formatName(name, batch) {
+  if (['predictText', 'predictImage'].indexOf(name) !== -1) {
+    name = 'apis';
+  }
+
+  if (batch) {
+    name += '/batch'
+  }
+
+  name = name.toLowerCase();
+  return name;
+}
+
+function postprocess(results, name, nameMap) {
+  if (name.indexOf('apis') === 0) {
```

Comment:
This function takes too many arguments, but I can't think of a great way to change that.